luigi/worker.py

```diff
@@ -490,9 +556,16 @@ def _get_work(self):
 
     def _run_task(self, task_id):
         task = self._scheduled_tasks[task_id]
-        p = TaskProcess(task, self._id, self._task_result_queue,
-                        random_seed=bool(self.worker_processes > 1),
-                        worker_timeout=self.__worker_timeout)
+
+        if task.run == NotImplemented:
+            p = ExternalTaskProcess(task, self._id, self._task_result_queue,
+                                    random_seed=bool(self.worker_processes > 1),
+                                    worker_timeout=self.__worker_timeout)
```

Comment:
This code makes a worker use an ExternalTaskProcess to execute a task that the scheduler has marked as `runnable`, but which has no implemented `run` method (and therefore is assumed to be an ExternalTask).

An ExternalTaskProcess is very similiar to a TaskProcess, but instead of calling the task's `run` method, it checks its `complete` method and returns a status based on that. If the task's `complete` method returns `False`, the task is marked as FAILED. The scheduler will then reschedule it later, subject to the normal retry logic (`disable-num-failures` especially applies here). 

So no, that code doesn't mean it will only be run once at the start of the pipeline (perhaps you're thinking of `check_complete` in worker.py?). The ExternalTaskProcess will be re-evaluated every time the task is given to a worker to execute.